### PR TITLE
Jon J jquery ajax daignostic

### PR DESCRIPTION
**Pull request by modularjon**
answer all questionss

### CODE REVIEW

**Comment by modularjon** [comment]
I forgot the  return on  $.ajax for delete donut.

File: diagnostic.md [hunk 1]
```diff
@@ -21,13 +21,18 @@ Write the `curl` request you'd use to retrieve a list of all donuts on the
 server.
 
 ```sh
-# your answer here
+curl --include --request "GET" "http://www.example.com/donuts"
```
Comment (berziiii) [inline diagnostic.md:24]: Correct.

File: diagnostic.md [hunk 2]
```diff
@@ -21,13 +21,18 @@ Write the `curl` request you'd use to retrieve a list of all donuts on the
...
 ```
 
 Write an AJAX request to retrieve a list of all donuts on the server.
 
 ```js
-let getDonuts = /* your answer here */;
+let getDonuts = function () {
+  return $.ajax({
```
Comment (berziiii) [inline diagnostic.md:31]: Correct.

File: diagnostic.md [hunk 3]
```diff
@@ -36,13 +41,18 @@ Now, we want to get a single donut from the server. Write the `curl` request
 you'd use to retrieve a single donut, using whatever ID you'd like.
 
 ```sh
-# your answer here
+curl --include --request "GET" "http://www.example.com/donuts/54"
```
Comment (berziiii) [inline diagnostic.md:44]: Correct.

File: diagnostic.md [hunk 4]
```diff
@@ -36,13 +41,18 @@ Now, we want to get a single donut from the server. Write the `curl` request
...
 ```
 
 Write an AJAX request to retrieve a single donut from the server.
 
 ```js
-let getDonut = /* your answer here */;
+let getDonut = function (donutID) {
```
Comment (berziiii) [inline diagnostic.md:50]: Correct.

File: diagnostic.md [hunk 5]
```diff
@@ -51,13 +61,18 @@ Write the `curl` request you'd use to delete a single donut, using whatever ID
 you'd like.
 
 ```sh
-# your answer here
+curl --include --request "DELETE" "http://www.example.com/donuts/99"
```
Comment (berziiii) [inline diagnostic.md:64]: Correct.

File: diagnostic.md [hunk 6]
```diff
@@ -51,13 +61,18 @@ Write the `curl` request you'd use to delete a single donut, using whatever ID
...
 ```
 
 Write an AJAX request to delete a single donut from the server.
 
 ```js
-let deleteDonut = /* your answer here */;
+let deleteDonut = function (donutID) {
+  $.ajax({
+    url: 'http://www.example.com/donuts/' + donutsID,
```
Comment (berziiii) [inline diagnostic.md:72]: Correct, watch spelling on `donutID` and `donutsID`

File: diagnostic.md [hunk 7]
```diff
@@ -73,13 +88,31 @@ data in JSON format.
...
 ```sh
-# your answer here
+curl --include --request "POST" "http://www.example.com/donuts/$ID" \
+  --header "Content-Type: application/json" \
+  --data "{
+    \"donut\" : {
+      \"name\": \"French Cruller\",
+      \"price\": \"\$0.99\"
+    }
+  }"
```
Comment (berziiii) [inline diagnostic.md:98]: Correct.

File: diagnostic.md [hunk 8]
```diff
@@ -73,13 +88,31 @@ data in JSON format.
...
 ```
 
 Write an AJAX request to create a single donut on the server using JSON.
 
 ```js
-let createDonut = /* your answer here */;
+let createDonut = function (donutID) {
```
Comment (berziiii) [inline diagnostic.md:104]: Remember, when you create a new `donut` you don't have an id. the ID gets assigned through the `POST` action.

File: diagnostic.md [hunk 9]
```diff
@@ -98,11 +131,27 @@ JSON.
...
+  return $.ajax({
+    url: 'http://www.example.com/donuts/' + donutID,
+    method: 'PATCH',
+    data: {
+      donut: {
+        name: "Krüller",
+      }
+    }
+  });
+};
```
Comment (berziiii) [inline diagnostic.md:156]: Correct.

File: diagnostic.md [hunk 10]
```diff
@@ -98,11 +131,27 @@ JSON.
...
 
 ```sh
-# your answer here
+curl --include --request "PATCH" "http://www.example.com/donuts/42" \
+  --header "Content-Type: application/json" \
+  --data "{
+    \"donut\": {
+      \"name\": \"Krüller\",
+    }
+  }"
```
Comment (berziiii) [inline diagnostic.md:140]: Correct.

**Comment by berziiii** [comment]
9 / 10. Nice work!

**Pull request closed without merging**